src/bp-groups/bp-groups-functions.php

```diff
@@ -5230,3 +5245,95 @@ function bb_groups_settings_default_fallback( $setting_type, $val = '' ) {
 	 */
 	return apply_filters( 'bp_group_' . $setting_type . '_status_fallback', $val );
 }
+
+/**
+ * Run migration for resolving the issue related to the groups.
+ *
+ * @since BuddyBoss [BBVERSION]
+ */
+function bb_group_migration() {
+
+	// When 'group restrict invites' is on, remove subgroup members not in the parent group.
+	if (
+		true === bp_enable_group_hierarchies() &&
+		true === bp_enable_group_restrict_invites() &&
+		function_exists( 'bb_groups_migrate_subgroup_member' )
+	) {
+		bb_groups_migrate_subgroup_member();
+	}
+
+}
+
+/**
+ * Migrate subgroup members.
+ *
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @return void
+ */
+function bb_groups_migrate_subgroup_member() {
+	global $wpdb, $bb_background_updater;
+
+	$results = $wpdb->get_results( "SELECT * FROM {$wpdb->prefix}bp_groups g WHERE g.parent_id != 0" );
```

Comment:
Optimize SQL code which can directly return the group ID and user_id which have parent_id != 0 and user have member of the group.
Eg: `SELECT g.id, gm.user_id FROM `wp_bp_groups` g inner JOIN wp_bp_groups_members gm ON gm.group_id = g.id WHERE g.parent_id != 0 and gm.is_confirmed = 1;`